report/4.Design/Object_Localization.tex

```diff
@@ -0,0 +1,113 @@
+\section{Object localization}\label{Design:ObjectLocalization}
+An early working product was favored in order to get a baseline for all components to be further developed from.
+Therefore since object localization using machine learning algorithms requires a bit of setup, along with a lot of training data some algorithms were implemented without machine learning.
+
+\subsection{Zone Average}
+Initially an object detection algorithm dubbed "Zone Average" were created.
+Zone Average locates the most red object in an image.
+It works by splitting the image into $N\times N$ tiles, and determining the total redness within each tile.
+The tile that contains the most red is determined to contain the object, and the location of that tile is returned as a result.
+
+In order to increase accuracy, dynamic resizing of tiles were introduced.
+The idea is that if the same tile has been the selected tile for multiple iterations, the size of the tiles should decrease to get better accuracy.
+Likewise when the object begins moving the size of the tiles should expand.
+
+\figur{0.6}{images/zone_avg.png}{Zone average algorithm detecting red balloon.}{fig:zone_avg}
+
+Zone Average functioned well as a starting point, but it lacked accuracy as also illustrated in image \ref{fig:zone_avg} where multiple tiles could be very red, but not necessarily in the center.
+
+\subsection{Object Fill}
+Following Zone Average a new algorithm called "Object Fill" was created.
+Object Fills purpose was to render very accurate position results, with the initial limitation of only being able to locate a single red object.
```

Comment:
Isn't it technically just an object of a color? I've added two suggestions, just choose.
```suggestion
Object Fills purpose was to render very accurate position results, with the initial limitation of only being able to locate a single coloured object.
```